### PR TITLE
init:create project setup and file structure

### PR DESCRIPTION
**Pull request by UtsavK55**
Initialized project: Set up the project repository with essential configurations.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+dist
```
Comment (adiTrivsme) [inline .gitignore:1]: use `Prettier` to avoid these formatting issues

File: tsconfig.json
Comment (adiTrivsme) [inline tsconfig.json:1]: Keep only the options you need, this looks pretty cluttered

**Review by adiTrivsme — CHANGES_REQUESTED** [review]

File: tsconfig.json
Comment (UtsavK55) [reply tsconfig.json:1]: I have resolved. Please review it.

File: .gitignore (hunk 1 above)
Comment (UtsavK55) [reply .gitignore:1]: I have added prettier.

**Review by adiTrivsme — APPROVED** [review]

**Pull request merged**